src/analysis/convex/function.lean

```diff
@@ -28,264 +28,239 @@ a convex set.
 open finset linear_map set
 open_locale big_operators classical convex pointwise
 
-variables {𝕜 E F ι ι' β : Type*} [add_comm_group E] [module ℝ E] [add_comm_group F] [module ℝ F]
-  [ordered_add_comm_monoid β] [module ℝ β] {s : set E}
+variables {𝕜 E F β ι : Type*}
+
+section ordered_semiring
+variables [ordered_semiring 𝕜] [add_comm_monoid E] [add_comm_monoid F]
+
+section ordered_add_comm_monoid
+variables (𝕜) [ordered_add_comm_monoid β]
 
 /-- Convexity of functions -/
-def convex_on (s : set E) (f : E → β) : Prop :=
-  convex ℝ s ∧
-  ∀ ⦃x y : E⦄, x ∈ s → y ∈ s → ∀ ⦃a b : ℝ⦄, 0 ≤ a → 0 ≤ b → a + b = 1 →
+def convex_on [has_scalar 𝕜 E] [has_scalar 𝕜 β] (s : set E) (f : E → β) : Prop :=
```

Comment:
Given the size of this PR, I am interested in ways to break it down.

Would it be feasible to change only this definition, globally replace `convex_on` with `convex_on ℝ` everywhere, and likewise for `concave_on`?

Reply:
Yes, absolutely. Do you want me to do that?

Reply:
Yes please!